system/jhtree/jhtree.cpp

```diff
@@ -1949,56 +1658,304 @@ bool CKeyCursor::ltEqual(const char *src, char *dst, bool seekForward)
+                eof = true;
+            lscans++;
+        }
+        else
+        {
+            if (!gtEqual(keyBuffer, keyBuffer, stats))
+                eof = true;
+            lseeks++;
+        }
+        if (!eof)
+        {
+            unsigned i = 0;
+            matched = true;
+            if (segs->segMonitors.length())
+            {
+                for (; i <= lastSeg; i++)
+                {
+                    matched = segs->segMonitors.item(i).matchesBuffer(keyBuffer);
+                    if (!matched)
+                        break;
+                }
+            }
+            if (matched)
+            {
+                ret = true;
+                break;
+            }
+#ifdef  __linux__
+            if (linuxYield)
+                sched_yield();
+#endif
+            eof = !segs->incrementKey(i, keyBuffer);
+            if (!exact)
+            {
+                ret = true;
+                break;
+            }
+            lwildseeks++;
+        }
+        else
+            eof = true;
     }
-    else
-        node.clear();
+    if (logExcessiveSeeks && lwildseeks > 1000)
+        reportExcessiveSeeks(lwildseeks, lastSeg, stats);
+    stats.noteSeeks(lseeks, lscans, lwildseeks);
+    return ret;
 }
 
-const byte *CKeyCursor::loadBlob(unsigned __int64 blobid, size32_t &blobsize)
+bool CKeyCursor::lookupSkip(const void *seek, size32_t seekOffset, size32_t seeklen, KeyStatsCollector &stats)
 {
-    const byte *ret = key.loadBlob(blobid, blobsize);
-    activeBlobs.append(ret);
+    if (skipTo(seek, seekOffset, seeklen))
+        stats.noteSkips(1, 0);
+    else
+        stats.noteSkips(0, 1);
+    bool ret = lookup(true, stats);
+#ifdef _DEBUG
+    if (traceSmartStepping)
+    {
+        StringBuffer recstr;
+        unsigned i;
+        for (i = 0; i < key.keySize(); i++)
+        {
+            unsigned char c = ((unsigned char *) keyBuffer)[i];
+            recstr.appendf("%c", isprint(c) ? c : '.');
+        }
+        recstr.append ("    ");
+        for (i = 0; i < key.keySize(); i++)
+        {
+            recstr.appendf("%02x ", ((unsigned char *) keyBuffer)[i]);
+        }
+        DBGLOG("SKIP: Got skips=%02d seeks=%02d scans=%02d : %s", stats.skips, stats.seeks, stats.scans, recstr.str());
+    }
+#endif
     return ret;
 }
 
-void CKeyCursor::releaseBlobs()
+
+unsigned __int64 CKeyCursor::getCount(KeyStatsCollector &stats)
 {
-    ForEachItemIn(idx, activeBlobs)
+    reset();
+    unsigned __int64 result = 0;
+    unsigned lseeks = 0;
+    unsigned lastRealSeg = segs->lastRealSeg();
+    for (;;)
     {
-        free((void *) activeBlobs.item(idx));
+        if (_lookup(true, lastRealSeg, stats))
+        {
+            unsigned __int64 locount = getSequence();
+            endRange(lastRealSeg);
+            ltEqual(keyBuffer, stats);
+            lseeks++;
+            result += getSequence()-locount+1;
+            if (!incrementKey(lastRealSeg))
+                break;
+        }
```

Comment:
this also used to set matched to false.  Is that still needed?

Reply:
I don't believe so

Reply:
ltEqual clears it